starter-code/scripts/article.js

```diff
@@ -17,12 +17,14 @@ Article.prototype.toHtml = function() {
   //   The result is added to the object as a new property, which can then be referenced
   //   by a key in the template. For example, you might want to display how old a post is,
   //   or say "(draft)" if it has no publication date:
-  this.daysAgo = parseInt((new Date() - new Date(this.publishedOn))/60/60/24/1000);
+  this.daysAgo = parseInt((new Date() - new Date(this.publishedOn)) / 60 / 60 / 24 / 1000);
   this.publishStatus = this.publishedOn ? 'published ' + this.daysAgo + ' days ago' : '(draft)';
   // TODO: Use handlebars to render your articles!
   //       - Select your template from the DOM.
   //       - Now "compile" your template with Handlebars.
   //       - Don't forget to return your template for this article.
+  var blackMagicCompile = Handlebars.compile($('#articleTemplate').html());
+  $('#articles').append(blackMagicCompile(this));
```

Comment:
I love this -- blackMagicCompile! One thing to mention - you don't actually have to worry about appending your filled-in templates here. That work is being done for you further down the page in the forEach. 
```
articles.forEach(function(a){
  $('#articles').append(a.toHtml());
});
```

This line is taking every article in the article array, calling toHtml on it (effectively creating a compiled template which is returned), and then appending that template to the #articles section. Therefore, all you have to worry about it returning the compiled template in your toHtml method.